src/box/memtx_tx.c

```diff
@@ -2420,7 +2420,7 @@ memtx_tx_history_remove_added_story(struct txn_stmt *stmt)
 static void
 memtx_tx_history_remove_deleted_story(struct txn_stmt *stmt)
 {
-	memtx_tx_history_rollback_deleted_story(stmt);
+	memtx_tx_story_unlink_deleted_by(stmt->del_story, stmt);
```

Comment:
Thanks for the question, elaborated in commit message:
```
Since we delete stories on commit now, there can be concurrent
transactions that must be aborted as readers of prepared values - that's
what rollback of prepared statements did, and after the fix this logic
is gone, so the commit seems like regression. But anyway, such design is
bad and is going to be fixed in future commits - we will delete stories
only when there are no concurrent transactions so we won't need to abort
readers of preapared tuples at all.
```

Reply:
> Why are we doing this for all statements, not only for prepared ones?

In `memtx_tx_history_rollback_deleted_story` we abort readers of the story if it was prepared and then unlink it. The commit only removes the logic with aborting readers of prepared stories since this logic doesn't work correctly on space delete.

Reply:
> I don't like that we have a commit that introduces a regression, it can break bisection. Rather than adding comments, better squash it with the fix.

OK, I'll try to change structure of commits.

Reply:
Done, no regressions anymore.

Moreover, I've thoroughly described my approach in `memtx: re-design handling of DDL in memtx MVCC` to explain what the patch doing and why.